src/tools/netkey-tool.c

```diff
@@ -562,6 +565,7 @@ int main(
 
 	if((r = sc_lock(card))<0){
 		fprintf(stderr,"Lock failed: %s\n", sc_strerror(r));
+		sc_release_context(ctx);
```

Comment:
Here you need to call `sc_card_free(card)` too as `sc_connect_card()` allocates new `sc_card_t` structure.

Reply:
Initializing it to NULL on a line 442 is also a good idea just to be sure.

Reply:
Hello~ I followed your advice and submitted a new PR(https://github.com/OpenSC/OpenSC/pull/2138), but the LGTM analysis was not successful. I found that the `sc_card_free()` was not defined in src/tools/netkey-tool.c . Maybe you mean `sc_disconnect_card()` ?

Reply:
You don't need to submit a new PR -- you can push more commits to this one. Right. The `sc_card_free` is not public. `sc_disconnect_card` is the right function. Sorry for a confusion.

Reply:
I'm so sorry to disturb you again. I found that no `sc_disconnect_card` is called when connect-card failed in other scripts (e.g. `main` in src\tools\npa-tool.c), so I am still a little confused about calling `sc_disconnect_card` here.

Reply:
It is the way as @frankmorgner wrote in #2135 -- calling exit frees all the memory so making sure we free that explicitly is mostly a good custom here and protection on "bad" platforms that might not do that. For example, `opensc-tool` calls this in the only in the `end` block  (reached from all error condition using `goto`). This is similar in all the other tools.

Reply:
Thank you very much for taking the time to reply! I've pushed another commit to this PR~ (This was just my second attempt to submit a PR, and I did learn a lot. Thanks again!)

Reply:
You are welcomed. Change looks good. Thank you for your contribution!